cmd/kubeadm/app/apis/kubeadm/v1beta2/types.go

```diff
@@ -119,6 +119,12 @@ type ClusterConfiguration struct {
 
 // ControlPlaneComponent holds settings common to control plane component of the cluster
 type ControlPlaneComponent struct {
+	// RunAsUser is the UID that the component container should run as.
+	RunAsUser *int64 `json:"runAsUser,omitempty"`
```

Comment:
@astrieanna 
we do a lot of planning in terms of what goes in the kubeadm config as it touches a lot of users. :)
this was not discussed as part of the 1.15 cycle so we might have to hold the change until the next configuration iteration.

in terms of the configurability, this might be a step in the right direction, on *nix.
however int64, is not a portable type with regard to Windows SIDs (e.g. if windows supports this longterm).
golang's stdlib uses `string` https://golang.org/pkg/os/user/#User

/hold
/assign @rosti @timothysc

Reply:
We used int64 because that was the type in SecurityContext.

Reply:
`SecurityContext` was not designed with Windows in mind. i have not information if it will change to `string`, but most likely not. also i have no information when Windows will support this type of sandboxing (if ever).

but we should use `string` to be portable like the standard library.

Reply:
The control plane does not run on Windows and I don't believe there are any plans for it to ever run on Windows.